proposals/support_custom_ecs_fields.md

```diff
@@ -0,0 +1,125 @@
+
+*Pros*:
+* More streamlined user experience that would be more "ECS native"
+* De-couples us from docker-compose and libcompose
+* Would be a step towards the larger ECS Local DevX goal
+
+*Cons*:
+* May require a shim to continue working directly with docker-compose and maintain backwards compatibility
+* Would be difficult to integrate with libcompose, and possibly require implementing a different parsing tool.
+* May break existing workflows that use docker-compose
+* Broadens scope of current feature request
+
+## Issues
+
+We currently depend on [libcompose](https://github.com/docker/libcompose) to parse the Composefile. This dependency ties us to only being able to support versions of the Composefile that libcompose also supports.
+
+## Related work:
+* Create new API that allows converting an ECS Task Definiton (json or yaml) into a Docker Composefile
+* container-transform: allows converting Composefile to Task Def
+
+# Proposal: Option 2
+
+While Option 1 would likely be the fastest solution to achieve the desired functionality, its impact on the extensibility of the CLI made it not the ideal route. Option 3 is more aligned with the broader vision of designing a new local development experience using the ECS CLI, but was not compatible enough with existing workflows based on docker-compose and would have a much broader scope than just delivering the desired functionality.
+
+Currently, the local development experience of a large number of our customers is based around using docker-compose. We want to continue supporting customers who use docker-compose as they are using it today, while allowing ECS custom fields to be used in production. This keeps ECS fields separate from fields that can only really be tested locally, as the fields not currently supported are dependent on calling the ECS backend.
+
+While we currently provide the ability to pass in more than one compose file (cf: [PR#263](https://github.com/aws/amazon-ecs-cli/pull/263)), the use case for that feature was to allow for overrides to a base compose file in order for customers to more easily configure different environments. These overrides, however, were on fields only supported by Docker Compose. The fields we are trying to support here are specific to ECS Task Definitions. Passing them in through a separate file would allow customers to maintain the integrity of existing composefiles as well as clearly communicate any differences between Docker services and ECS container definitions. It would also allow for more modular validation of any ECS-specific data.
+
+## Implementation
+
+* Pass in a file with ECS task definition fields via a new flag, `--task-def-fields` (Other possible names: `--ecs-fields`, `ecs-task-def-fields`, `ecs-compose`, `--ecs-extensions`, `ecs-task-definition`).
```

Comment:
I personally like `--ecs-fields` and `--task-def-fields`, but all of these are fine.